### ISSUE TITLE
about wavenet mixture loss

### ISSUE THREAD

**Issue by azraelkuan** [twidddj/tf-wavenet_vocoder#3]
Hello, thanks to your great work!
I have seen your mixture code, how does the loss change in the training process?
I trained it in my vocoder project, but it can sample good $x$ in the sample code?
what do you changed? I have been in trobule with it for weeks.

**Comment by twidddj** [comment]
Hi @azraelkuan, I think you have another problem not in the mixture. You have same process as mine. Did you check the result of your model in teacher forcing mode? If it works, the generation part might have a problem. Otherwise, I'm not sure the model is learning properly. It might be required to check your upsampled Mel-spectrum. The Mel-spectrum may change like below images.

step 18K
![step18000](https://user-images.githubusercontent.com/2585210/38605632-dd4bb97e-3dae-11e8-8862-4ad9906e218e.png)

step 60K
![step60000](https://user-images.githubusercontent.com/2585210/38605664-ee5c7b2c-3dae-11e8-9b87-aebe4e958eaa.png)

step 430K
![step430000](https://user-images.githubusercontent.com/2585210/38605697-0498d2e6-3daf-11e8-9376-80f6a4707e67.png)

In my case, the loss starts over 100 and decreased under 60.

**Comment by azraelkuan** [comment]
### For mixture loss:
In fact, the mixture loss have problem, you should change the mixture loss function 
https://github.com/twidddj/tf-wavenet_vocoder/blob/0e3f00ac5d347cd8bf43db3d1830844b6413fb47/wavenet/mixture.py#L74 
to`log_probs = log_probs+ tf.nn.log_softmax(logit_probs, -1)` and do not use reduce_sum in the final, you will get the loss from about 10 to 6. 
### For upsampled mel sp
Do u use the transposed conv for upsample or just repeat? I test the upsample conv in the 8bit mode, and it works well. I don't check it in the teacher forcing mode, i just doubt that my generated code has some problem, if u have sample successful, then it will be another problem i don't find

**Comment by twidddj** [comment]
For mixture loss, you are right. It was wrong. Thanks to your kindness.

You'll find the method for upsampling in [this code](https://github.com/twidddj/tf-wavenet_vocoder/blob/master/apps/vocoder/model/vocoder.py) easily. 

Our wavenet model has started from [this project](https://github.com/twidddj/vqvae) which has built from ibab's wavenet. Some of them would be different of yours. You can find the differences in [this file](https://github.com/twidddj/tf-wavenet_vocoder/blob/master/wavenet/model.py). 

If you have any trouble on my work, feel free and let me know about it.
